### ISSUE TITLE
python `make` sprintf compilation error - Raspbian

### ISSUE THREAD

**Issue by julianrubisch** [libpd/libpd#244]
Currently, I'm trying to compile the python section of libpd on Raspbian Stretch.

`pi@raspberrypi:~/libpd/python $ make`

It gives me the following error(s):

```
../pure-data/src/x_connective.c:1337:9: error: format not a string literal and no format arguments [-Werror=format-security]
         sprintf(buf, x->x_format->s_name);
         ^~~~~~~
../pure-data/src/x_connective.c: In function 'makefilename_symbol':
../pure-data/src/x_connective.c:1376:9: error: format not a string literal and no format arguments [-Werror=format-security]
         sprintf(buf, x->x_format->s_name);
         ^~~~~~~
../pure-data/src/x_connective.c: In function 'makefilename_bang':
../pure-data/src/x_connective.c:1400:9: error: format not a string literal and no format arguments [-Werror=format-security]
         sprintf(buf, x->x_format->s_name);
         ^~~~~~~
At top level:
../pure-data/src/x_connective.c:171:13: warning: 'pdsymbol_list' defined but not used [-Wunused-function]
 static void pdsymbol_list(t_pdsymbol *x, t_symbol *s, int ac, t_atom *av)
             ^~~~~~~~~~~~~
cc1: some warnings being treated as errors
error: command 'arm-linux-gnueabihf-gcc' failed with exit status 1
Makefile:22: recipe for target 'build' failed
make: *** [build] Error 1
```

It seems as though this is related to the following PR on pure-data: https://github.com/pure-data/pure-data/pull/282/commits/382f703a3eeb5ca39839a4b29d19747dd11b6e6c

Is there any way how I can update the pure-data submodule? Or make this work in another way? I could probably cherry-pick this commit but I'm not used to working with submodules...

Interestingly, `make` of libpd itself (in the main directory) did work out fine.

**Comment by tkirshboim** [comment]
I believe that https://github.com/libpd/libpd/pull/243 is addressing the same issue.
@julianrubisch You could try and use my [fork of libpd](https://github.com/tkirshboim/libpd) which includes that 'pure-data' commit you're mentioning, just to validate that it also solves this issue.

**Comment by julianrubisch** [comment]
For the record, I temporarily solved this by checking out the above mentioned commit:

`pi@raspberrypi:~/libpd/python $ git checkout 382f703a3eeb5ca39839a4b29d19747dd11b6e6c`

@tkirshboim thanks for referencing the PR, I didn't search well enough, obviously :)
Can confirm that it works, though.

**Comment by danomatika** [comment]
This should be fixed by #243.
